server/services/carrier.js

```diff
@@ -333,8 +333,12 @@ module.exports = class CarrierService extends EventEmitter {
             }
 
             carrier.ownedByPlayerId = star.ownedByPlayerId; // Transfer ownership
-            carrier.specialistId = null; // Remove the specialist. Note that this is required to get around an exploit where players can use a gift just before a battle to weaken the opponent.
-
+            
+            // Remove the specialist. Note that this is required to get around an exploit where players can use a gift just before a battle to weaken the opponent.
+            if (!this.diplomacyService.getDiplomaticStatusBetweenPlayers(game, [carrierPlayer,starPlayer]) == 'allies') {
```

Comment:
Nearly.

Need to consider whether formal alliances are **disabled**, it should set the `specialistId` to `null` as it did before.

```js
if (!this.diplomacyService.isFormalAlliancesEnabled(game) || !this.diplomacyService.isDiplomaticStatusBetweenPlayersAllied(game, [carrierPlayer._id, starPlayer._id]) { 
    carrier.specialistId = null;
}
```

I think that should do it.

Reply:
Requested change applied